applications/FluidDynamicsApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,432 @@
+{
+    return Kratos::make_intrusive<QSVMSDEMCoupled>(NewId, this->GetGeometry().Create(ThisNodes), pProperties);
+}
+
+template< class TElementData >
+Element::Pointer QSVMSDEMCoupled<TElementData>::Create(IndexType NewId,GeometryType::Pointer pGeom,Properties::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<QSVMSDEMCoupled>(NewId, pGeom, pProperties);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Inquiry
+
+template< class TElementData >
+int QSVMSDEMCoupled<TElementData>::Check(const ProcessInfo &rCurrentProcessInfo)
+{
+    int out = QSVMS<TElementData>::Check(rCurrentProcessInfo);
+    KRATOS_ERROR_IF_NOT(out == 0)
+        << "Error in base class Check for Element " << this->Info() << std::endl
+        << "Error code is " << out << std::endl;
+
+    for(unsigned int i=0; i<NumNodes; ++i)
+    {
+        Node<3>& rNode = this->GetGeometry()[i];
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(ACCELERATION,rNode);
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(NODAL_AREA,rNode);
+    }
+
+    return out;
+}
+
+template <class TElementData>
+void QSVMSDEMCoupled<TElementData>::Calculate(
+    const Variable<double>& rVariable,
+    double& rOutput,
+    const ProcessInfo& rCurrentProcessInfo)
+    {
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+
+        if (rVariable == ERROR_RATIO)
+        {
+            const double density = this->GetAtCoordinate(data.Density,data.N);
+
+            // Get Advective velocity
+            array_1d<double, 3> convective_velocity= this->GetAtCoordinate(data.Velocity, data.N) -
+            this->GetAtCoordinate(data.MeshVelocity, data.N);
+
+            // Output container
+            array_1d< double, 3 > ElementalMomRes(3, 0.0);
+
+            // Calculate stabilization parameter. Note that to estimate the subscale velocity, the dynamic coefficient in TauOne is assumed zero.
+            double tau_one, tau_two;
+            this->CalculateTau(data, convective_velocity, tau_one, tau_two);
+
+            if ( data.UseOSS != 1 ) // ASGS
+            {
+                this->AlgebraicMomentumResidual( data, convective_velocity, ElementalMomRes);
+                ElementalMomRes *= tau_one;
+            }
+            else // OSS
+            {
+                this->OrthogonalMomentumResidual(data, convective_velocity, ElementalMomRes);
+                ElementalMomRes *= tau_one;
+            }
+
+            // Error estimation ( ||U'|| / ||Uh_gauss|| ), taking ||U'|| = TauOne ||MomRes||
+            double ErrorRatio(0.0);//, UNorm(0.0);
+
+            for (unsigned int i = 0; i < Dim; ++i)
+            {
+                ErrorRatio += ElementalMomRes[i] * ElementalMomRes[i];
+            }
+            ErrorRatio = sqrt(ErrorRatio); // / UNorm);
+            ErrorRatio /= density;
+            this->SetValue(ERROR_RATIO, ErrorRatio);
+            rOutput = ErrorRatio;
+        }
```

Comment:
Did you check the `ERROR_RATIO` calculation? This implementation comes from the old VMS element and I'm afraid it is not fully compatible. 

Aside of this, this could be implemented in the base QS class.